src/Page/QuestionDetailPage/index.tsx

```
@@ -16,45 +16,57 @@ import {
 import { useAuthStore } from '../../hooks/useStore';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import baseFontStyle from '../../styles/font.css';
+import { problemApiWrapper } from '../../api/wrapper/problem/problemApiWrapper';
+import { URL, URLWithParam } from '../../constants/url';
+import { IProblemDetailResponseData } from '../../types/api/problem';
 
 function QuestionDetailPage() {
-  const state = useLocation().state as IProblemIdLinkState;
-  const problemData = listData[state.problemId];
-
+  const { id } = useParams();
   const { isLogin } = useAuthStore();
+  const [data, setData] = useState<IProblemDetailResponseData>();
+
   const [isDark, setIsDark] = useState(false);
 
   function toggleDarkMode() {
     setIsDark((prev) => !prev);
   }
 
+  function handleSubmit() {
+    return;
+  }
+
+  useEffect(() => {
+    if (!id) return;
+    problemApiWrapper.problemDetail(id).then((res) => {
+      setData(res.data);
+    });
+  }, []);
+
   return (
     <>
       <Header />
       <main className={`${isDark ? themeDarkClass : themeLightClass} ${pageStyle}`}>
         <div className={topStyle}>
           <div className={descStyle}>
             <div className={titleTagStyle}>
-              <h1 className={baseFontStyle.title}>{problemData.title}</h1>
+              <h1 className={baseFontStyle.title}>{data?.title}</h1>
               <ul>
-                {problemData.tagList.map((tagName) => (
+                {data?.tags.map((tagName) => (
                   <Tag name={tagName} key={tagName} />
                 ))}
               </ul>
             </div>
             <div className={baseFontStyle.medium}>
-              {`제출 : ${problemData.numberSolved}, 평균 점수 : ${problemData.averageScore}점, 최고점 : ${problemData.highestScore}점 , 최저점 : ${problemData.lowestScore}점`}
+              {`제출 : ${data?.totalSolved}, 평균 점수 : ${data?.avgScore}점, 최고점 : ${data?.topScore}점 , 최저점 : ${data?.bottomScore}점`}
```

전반적으로 optional chaining시 기본 값이 설정이 안되어 있는 것 같아요.

---

위 코드에서 지운 부분은 mock data로 값이 있는것이 확정적이고 아래는 API 호출을 통해 얻은 데이터라 optional chaining이 필요했습니다. view에 보여지는데 값이 없고(null or undefined) 그것이 허용되는  경우에는 0을 보여주도록 하고, 값이 꼭 있어야 하는 경우에는 없으면 return 하도록 했습니다. 추후 error page를 만들어 해당 페이지로 이동하도록 처리하겠습니다~!!